### PR TITLE
feat(swarm)!: report connections to our own `PeerId` in separate error

### PR DESCRIPTION
**Pull request by melekes**
## Description

Previously, inbound connections that happened to resolve to our own `PeerId` were reported as `WrongPeerId`. With this patch, we now report those in a dedicated `LocalPeerId` error.

Related: #3205.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,8 @@
   If you have previously set `connection_event_buffer_size` you should re-evaluate what a good size for a _per connection_ buffer is.
   See [PR 3188].
 
+- Add `PendingConnectionError::LocalPeerId` to differentiate wrong VS local peer ID errors. See [PR 3377]
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md]:
```suggestion
- Add `PendingConnectionError::LocalPeerId` to differentiate wrong VS local peer ID errors. See [PR 3377].
```

File: swarm/src/lib.rs [hunk 2]
```diff
@@ -1581,7 +1581,7 @@ pub enum DialError {
     /// has been reached.
     ConnectionLimit(ConnectionLimit),
     /// The peer being dialed is the local peer and thus the dial was aborted.
```
Comment (thomaseizinger) [inline swarm/src/lib.rs]: These docs need updating now because we are now also using this variant for inbound connections.

**Review by thomaseizinger — COMMENTED** [review]
Thanks!

One comment, plus `libp2p-metrics` needs updating.

File: swarm/src/lib.rs (hunk 2 above)
Comment (melekes) [reply swarm/src/lib.rs:1583]: But this is `DialError`

**Review by thomaseizinger — DISMISSED** [review]

**Comment by thomaseizinger** [comment]
Thank you!

**Comment by thomaseizinger** [comment]
@mergifyio refresh

**Review by thomaseizinger — APPROVED** [review]

**Comment by mxinden** [comment]
Thanks @melekes!

**Pull request merged**